examples/all-clusters-app/linux/main-common.cpp

```diff
@@ -88,27 +94,78 @@ namespace {
 // This file is being used by platforms other than Linux, so we need this check to disable related features since we only
 // implemented them on linux.
 #if CHIP_DEVICE_LAYER_TARGET_LINUX
+constexpr EndpointId kNetworkCommissioningEndpointMain      = 0;
+constexpr EndpointId kNetworkCommissioningEndpointSecondary = 0xFFFE;
+
 #if CHIP_DEVICE_CONFIG_ENABLE_THREAD
 NetworkCommissioning::LinuxThreadDriver sLinuxThreadDriver;
-Clusters::NetworkCommissioning::Instance sThreadNetworkCommissioningInstance(0 /* Endpoint Id */, &sLinuxThreadDriver);
+Clusters::NetworkCommissioning::Instance sThreadNetworkCommissioningInstance(kNetworkCommissioningEndpointMain,
+                                                                             &sLinuxThreadDriver);
 #endif
 #if CHIP_DEVICE_CONFIG_ENABLE_WPA
 NetworkCommissioning::LinuxWiFiDriver sLinuxWiFiDriver;
-Clusters::NetworkCommissioning::Instance sWiFiNetworkCommissioningInstance(1 /* Endpoint Id */, &sLinuxWiFiDriver);
-#endif
+Clusters::NetworkCommissioning::Instance sWiFiNetworkCommissioningInstance(kNetworkCommissioningEndpointSecondary,
+                                                                           &sLinuxWiFiDriver);
 #endif
+#endif // CHIP_DEVICE_LAYER_TARGET_LINUX
 } // namespace
 
 void ApplicationInit()
 {
+    // Enable secondary endpoint only when we need it, this should be applied to all platforms.
+    emberAfEndpointEnableDisable(kNetworkCommissioningEndpointSecondary, false);
+
 #if CHIP_DEVICE_LAYER_TARGET_LINUX && defined(ZCL_USING_LEVEL_CONTROL_CLUSTER_SERVER)
```

Comment:
Why do we care about LEVEL_CONTROL_CLUSTER here?

And the definition of kNetworkCommissioningEndpointSecondary needs to move out of the CHIP_DEVICE_LAYER_TARGET_LINUX ifdef above....

Reply:
Not sure, seems it exists long ago, removed